### ISSUE TITLE
Decimal values are rounded to js-Number precision

### ISSUE THREAD

**Issue by SergeySeleznyov** [aspnet/Blazor#695]
Let's take the "FetchData.cshtml" example form the project template.

Modify the data type:

```
public class WeatherForecast
{
    ... // origin properties (not changed)
    public decimal DecimalProp { get; set; }
    public string StringDecimalProp { get { return DecimalProp.ToString(); } }
}

```

JSON that comes from the server to the client may look as follows:

```
[{
	...
	       DecimalProp:16808220559571622000
	StringDecimalProp:"16808220559571621480.16240545"
},
{
	...
	       DecimalProp:-54580.79494236794
	StringDecimalProp:"-54580.794942367938243476284644"
}...]

```

Thus, on the client side, WeatherForecast's "DecimalProp" property has a decimal type but the js-Number type's precision.

This behavior may be OK for the javascript processing, but in the case of Blazor, customers may expect passing server-side data and data-processing logic to the "client side" without JS-based side effects like losing precision. 

Is it going to be improved in the future?
Is there any doc with Blazor's data-passing limitation?

**Comment by Andrzej-W** [comment]
@SteveSandersonMS We would like to port our desktop LOB applications to the web. We ordinarily work with monetary values which have 14 significant digits (12 digits to the left and 2 to the right of decimal point). Sometimes we need 4 digits to the right and I believe that big companies often need more than 12 digits to the left of decimal point. In accounting systems accuracy is critical. It is not priority now but in my opinion this problem should be solved before Blazor 1.0.

**Comment by SergeySeleznyov** [comment]
@SteveSandersonMS, for a web developer such a restriction is natural and even predictable.
However, web developers have lots of libs and frameworks.
 
We are researching the possibility to build not an application but a tool set for the Blazor. We already see some desktop developers' interest in using Blazor to migrate their desktop-based client applications' data and logic to the browser. Please note that their data-types can be generated by some ORM-sort tooling. And it is completely unexpected for a desktop developer that the currency counting involving decimal data implicitly loses precision. 
 
I agree with @Andrzej-W that we can proceed working with Blazor keeping in mind that this behavior will be improved in the future.

**Comment by danroth27** [comment]
>  serializing the accuracy-critical properties as strings...should be possible to configure a more advanced JSON parser to do that for you.

@JamesNK Is precision handling of decimal values something that Json.NET tries to handle?

**Comment by JamesNK** [comment]
Yes. When deserializing to a decimal (the serializer sees the property is a decimal type) then the value will be deserialized without losing precision.

**Comment by danroth27** [comment]
Presumably precision is also preserved on serialization as well? Does Json.NET do this by serializing decimals as strings and then converting back to decimals on deserialization?

**Comment by JamesNK** [comment]
Serialization is easy - at a very simplified level it just involves calling ToString on both.
